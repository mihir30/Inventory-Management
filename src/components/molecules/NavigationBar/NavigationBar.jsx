import React from 'react'
import { NavLink } from 'react-router-dom'
import Button from '../../atoms/Button/Button'
import "./NavigationBar.scss"

const NavigationBar = () => {
    return (
        <nav className='navigation-bar'>
            <Button buttonType={"flex-sm-fill text-sm-center report-header-tile"}><NavLink className='navbar-link nav-link' to="/statistics">Statistics</NavLink></Button>
            <Button buttonType={"flex-sm-fill text-sm-center report-header-tile"}><NavLink className='navbar-link nav-link' to="/unitsbystate">Units by state</NavLink></Button>
            <Button buttonType={"flex-sm-fill text-sm-center report-header-tile"}><NavLink className='navbar-link nav-link' to="/integrity">Integrity</NavLink></Button>
            <Button buttonType={"flex-sm-fill text-sm-center report-header-tile"}><NavLink className='navbar-link nav-link' to="/topusers">Top users</NavLink></Button>
        </nav>
    )
}

export default NavigationBar