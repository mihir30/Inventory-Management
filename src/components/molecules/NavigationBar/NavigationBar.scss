.navigation-bar{
    border-radius: 2px 2px 0 0;
    box-sizing: border-box;
    cursor: pointer;
    display: table-cell;
    overflow: hidden;
    text-align: center;
    vertical-align: middle;
    background-color: #fff;
    display: flex;
    width: 80vw;
    border-bottom:1px solid #ccd7e6;
}
.report-header-tile{
    border: 0;
    color: #1ca1c1;
    font-size: 1pc;
    font-weight: 550;
    letter-spacing: 0;
    width:175px;
    background-color: #fff;
    &:active{
        box-shadow: inset 0 -2px #1ca1c1;
    }
    &:hover{
        background-color: #f4f5f9;
    }
}

.navbar-link{
    text-decoration: none;
    color: #1ca1c1;
    &:hover{
       color: #1ca1c1;
    }
    &:visited{
        color: #1ca1c1;
    }
    &:active{
        box-shadow: inset 0 -2px #1ca1c1;
    }
}

@media screen and (max-width: 768px) and (min-width:375px) {
    .navigation-bar{
        width: 105vw;
    }
   
}

@media screen and (max-width: 375px) {
    .navigation-bar{
        width: 220vw;
    }
   
}


