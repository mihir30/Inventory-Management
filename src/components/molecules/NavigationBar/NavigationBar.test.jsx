import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import NavigationBar from './NavigationBar';

describe("Navigation bar Component", () => {
    it('renders the component properly', () => {
        const divEl = document.createElement("div");
        ReactDom.render(<BrowserRouter><NavigationBar></NavigationBar></BrowserRouter>, divEl); 
        
    });


    it('should have the right navigation bar component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});