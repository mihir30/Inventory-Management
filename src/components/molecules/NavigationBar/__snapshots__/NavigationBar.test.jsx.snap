// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar Component should have the right navigation bar component snapshot 1`] = `null`;
